vncviewer/GestureHandler.cxx

```diff
@@ -150,40 +154,15 @@ void GestureHandler::getAverageDistance(double *first, double *last)
     *last = ld / size;
 }
 
-void GestureHandler::touchTimeout() {
+void GestureHandler::longpressTimeout() {
   // Have other updates already determined the gestures?
   if (hasDetectedGesture())
     return;
 
-  switch (tracked.size()) {
-    case 0:
-      this->state = GH_INITSTATE;
-      break;
-
-    case 1:
-      // Not a multi-touch event
-      this->state &= ~(GH_VPAN | GH_HPAN | GH_PINCH);
-      this->state &= ~(GH_1XTOUCH | GH_3XTOUCH);
-      break;
-
-    case 2:
-      // Not a single- or triple-touch gesture
-      this->state &= ~(GH_1XTOUCH | GH_3XTOUCH);
-      break;
-
-    case 3:
-      // Not a single- or double-touch gesture
-      this->state &= ~(GH_VPAN | GH_HPAN | GH_PINCH);
-      this->state &= ~(GH_1XTOUCH | GH_2XTOUCH);
-      break;
-
-    default:
-      this->state = GH_NOGESTURE;
-  }
-
-  if (hasDetectedGesture()) {
-    if ((tracked.size()) != 2 || (this->state != GH_2XTOUCH))
-      pushEvent(GH_GestureBegin);
+  if (tracked.size() == 1) {
+    // Emulate 2XTOUCH on single-touch long-press
+    this->state = GH_2XTOUCH;
```

Comment:
`2XTOUCH` is a lot better than "right button", but it is still a rather confusing API that returns "two finger touch" for what's actually just one finger. :)

Reply:
Open to suggestions...

Reply:
This was fixed with the new abstractions.